### ISSUE TITLE
Merge some review tools queues

### ISSUE THREAD

**Issue by wagnerand** [mozilla/addons-server#19984]
Merge _Recommended_, _Other Pending Review_ and _Unlisted Add-ons Pending Manual Approval_ queues into a new _Manual Review_ queue.

Q: Would we show the maliciousness score ?
A: Yes, if possible. We can consider not to if it’s a problem.

Q: For add-ons with multiple unlisted versions waiting for approval, what would we show ? Would we show one row per version, or per add-on ?
A: One row per add-on.

Q: For add-ons with a listed version and one or more unlisted versions waiting for approval, what would we show and what would we link to ?
A: We show one row, we link to listed.

Q: Do we need to show the version numbers ?
A: No.

@diox's assessment: "FWIW I think from an engineering standpoint, if we don’t care about showing version number, and only show the add-on, merging should be doable within the current architecture, so we could try that quickly without rewriting the whole thing. Maybe separately (keeping the existing queues for now) to experiment and iterate on it. Not entirely sure how that’d work with version-based flags that can be shown*, and also unsure about performance."

*Version-based flags is currently "sources provided", which we can drop from showing in the queue.

**Comment by eviljeff** [comment]
> Merge _Recommended_, _Other Pending Review_ and _Unlisted Add-ons Pending Manual Approval_ queues into a new _Manual Review_ queue.

These queues have different permissions.  Can we drop the extra permissions?  It's not _impossible_ to have a queue that shows different things based on permissions - we do it with `admin_reviewer` - but it makes the code to build the list of add-ons more complicated.  And operationally, different people having different combinations of permissions (and thus different queue views and counts) can add unnecessary complexity.

**Comment by wagnerand** [comment]
> > Merge _Recommended_, _Other Pending Review_ and _Unlisted Add-ons Pending Manual Approval_ queues into a new _Manual Review_ queue.
> 
> These queues have different permissions. Can we drop the extra permissions? It's not _impossible_ to have a queue that shows different things based on permissions - we do it with `admin_reviewer` - but it makes the code to build the list of add-ons more complicated. And operationally, different people having different combinations of permissions (and thus different queue views and counts) can add unnecessary complexity.

Yes, we can simplify the permissions for this queue. What exactly do you have in mind?

**Comment by eviljeff** [comment]
> Yes, we can simplify the permissions for this queue. What exactly do you have in mind?

the easiest solution would be to drop all the additional reviewer permissions and just rely on the `Addons:Review` permission.  If you think we still need separate permissions, maybe a `Addons:ManualReview" permission that would be needed to approve any add-on that wasn't already approved?

**Comment by diox** [comment]
The problem with simplifying permissions though, is that we can end up linking to review pages that would need a permission the user doesn't have. So we could:
- a) Ignore this problem for now
- b) Also refactor permissions required on review pages as well. Which raises more questions:
  - Just use `Addons:Review` everywhere ? Or another permission like @eviljeff pointed out ?
  - Do we replace all permission logic in the review page, or add that new one as an alternative ?

**Comment by wagnerand** [comment]
> The problem with simplifying permissions though, is that we can end up linking to review pages that would need a permission the user doesn't have.

Do you mean the user wouldn't have access to submit a review or the user wouldn't have access to even just loading the page?

**Comment by diox** [comment]
At the moment, if the reviewer doesn't have `ReviewerTools:ViewUnlisted` or `Addons:ReviewUnlisted`, they'll get a 403 just loading the unlisted review page, or the listed review page of a pure unlisted add-on.

On top of that of course, if they don't have the necessary permission they might be able to access the page but won't be able to do much on it.

**Comment by wagnerand** [comment]
Ok, that's fine for the moment. Everyone should have either of those. I'd suggest ignoring this for now and cleaning up the permissons later to unblock this issue.

**Comment by diox** [comment]
So the merged queue should use `Addons:Review` ?

**Comment by wagnerand** [comment]
> So the merged queue should use `Addons:Review` ?

Do you mean accessing that queue page itself? For that we can use `is_listed_addons_viewer_or_reviewer or is_unlisted_addons_viewer_or_reviewer`.

**Comment by wagnerand** [comment]
Something I just noticed: If the perf hit is affordable, the queue page should show the entries count. We could show it in the horizontal queues navigation bar or as a separate paragraph below the queue, for example.

**Comment by diox** [comment]
> Something I just noticed: If the perf hit is affordable, the queue page should show the entries count. We could show it in the horizontal queues navigation bar or as a separate paragraph below the queue, for example.

At all times or just when there is pagination ?

**Comment by wagnerand** [comment]
> > Something I just noticed: If the perf hit is affordable, the queue page should show the entries count. We could show it in the horizontal queues navigation bar or as a separate paragraph below the queue, for example.
> 
> At all times or just when there is pagination ?

I'd prefer at all times. The current page size is 100, which is fairly large to visually count with your eyes quickly.

**Comment by diox** [comment]
If I just display the paginator at all times, here is how it looks:
![Screenshot 2023-01-11 at 16-18-13 Reviewer Tools – Add-ons for Firefox](https://user-images.githubusercontent.com/187006/211844039-85d4392e-6454-4107-92bf-5cb7471dd584.png)

Note the lone `1` on the left - it's a link to the current page. If there were more pages we'd display more pagination links next to it. The result count is displayed on the right.

Is that ok or do you need more tweaks ?

**Comment by wagnerand** [comment]
That looks ok to me. Just for confirmation, assuming page size is 100 and there are 105 add-ons in the queue in total, the label would say "Results 1-100 of 105", right? If so, that's great.

**Comment by diox** [comment]
Yes.

**Comment by diox** [comment]
Compared to the description, the implementation:
- Doesn't show maliciousness score yet. That's https://github.com/mozilla/addons-server/issues/20220
- Doesn't necessarily show the listed version if there are multiple versions in different channels waiting for a manual review, instead we show (including version number) the one that has been waiting for a review the longest (oldest due date), and link to the review page for that channel.

**Comment by ioanarusiczki** [comment]
@diox I started testing around on AMO dev.

**Comment by ioanarusiczki** [comment]
@diox @wagnerand 

My notes after looking on -dev and -stage:

- addons from Manual Queue all have number versions next to the addon name.
- an addon with sources provided and no other flag -> for the listed channel will show up in auto-approved queue (after automatic approval). For unlisted -> addon shows up in All unlisted queue. I hope that's the expected result.
- delayed indefinitely by scanner addons will be displayed in the Manual  Review queue. 
- after a reviewer is manually disabling listed/unlisted auto-approval, an addon is displayed here.
- the recommended addons and notable addons are displayed here.
- with multiple versions awaiting review , listed and unlisted -> the first version submitted and awaiting review is displayed (no matter what the channel is). After approving the oldest the next version awaiting review is displayed. 
- pagination is 200 entries/page (seems to be 200 everywhere) 
- permissions: with admin rights all operations are available. With Addons:Review permission, the reviewer can approve/reject addons delayed by scanners or notable. These reviewers cannot approve/reject recommended and unlisted addons.
- the list on AMO stage displays many incomplete addons, I've also noticed one [invisible](https://addons.allizom.org/en-US/developers/addon/testrelease03-29-17-1-0/versions)
- on AMO dev: promoted addons other than recommended, such as spotlight, verified/sponsored (and I think that's expected) but also some automatically approved addons which seem to be submitted through the API. I'll give some examples:
https://reviewers.addons-dev.allizom.org/en-US/reviewers/review/624412
https://reviewers.addons-dev.allizom.org/en-US/reviewers/review-unlisted/624447
https://reviewers.addons-dev.allizom.org/en-US/reviewers/review/624410

**Comment by diox** [comment]
- Incomplete add-ons listed versions shouldn't appear in the queue, that is handled by a follow-up issue: https://github.com/mozilla/addons-server/issues/20198(though we didn't commit a migration to remove add-ons, if there are any)
- That last point needs more investigation. The due date should have been removed for those at approval, removing them from the queue.

**Comment by wagnerand** [comment]
> after a reviewer is manually disabling listed/unlisted auto-approval, an addon is displayed here.

You mean when a reviewer disabled auto-approval and the developer submits the next version afterwards, then it's shown in the queue, right?

**Comment by ioanarusiczki** [comment]
@wagnerand 

> You mean when a reviewer disabled auto-approval and the developer submits the next version afterwards, then it's shown in the queue, right?

Yes.

**Comment by eviljeff** [comment]
>     * That last point needs more investigation. The due date should have been removed for those at approval, removing them from the queue.

I'm unclear if we're talking about promoted groups that get auto-approved or not.  For auto approved groups (`pre_review = False`) there shouldn't be a `due_date` set for add-ons in the first place.  For those add-ons in groups that have pre review then yeah, approving (or rejecting) should have cleared them.

**Comment by diox** [comment]
https://reviewers.addons-dev.allizom.org/en-US/reviewers/review/624412 for instance doesn't appear to have anything special - it was auto-approved, not part of a promoted group, no flags set, yet its current version has a due date for some reason.

**Comment by KevinMind** [comment]
Old Jira Ticket: https://mozilla-hub.atlassian.net/browse/ADDSRV-128
